src/bun.js/webcore/TextDecoder.zig

```diff
@@ -280,46 +280,42 @@ fn decodeSlice(this: *TextDecoder, globalThis: *JSC.JSGlobalObject, buffer_slice
 }
 
 pub fn constructor(globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) bun.JSError!*TextDecoder {
-    var args_ = callframe.arguments_old(2);
-    var arguments: []const JSC.JSValue = args_.ptr[0..args_.len];
+    const arguments = callframe.argumentsAsArray(2);
+    const encoding_value = arguments[0];
+    const options_value = arguments[1];
```

Comment:
zig allows destructuring arrays, so this can be changed to
```suggestion
    const encoding_value, const options_value = callframe.argumentsAsArray(2);
```

Reply:
Oh that's pretty cool! I'll have to look a little more into that